Added CRD and design documents

shtripat opened the pull request:
monstorak-issue-id: monstorak/monstorak/issues/6
monstorak-issue-id: monstorak/monstorak/issues/22
Signed-off-by: Shubhendu <shtripat@redhat.com>

Commit:
Added CRD and design documents

monstorak-issue-id: monstorak/monstorak/issues/6
monstorak-issue-id: monstorak/monstorak/issues/22
Signed-off-by: Shubhendu <shtripat@redhat.com>

shtripat commented:
@aruniiird @umangachapagain please review

umangachapagain commented:
monstorak/pull/25 Captures my thought process. We could combine both.

shtripat commented:
> ures my thought process. We could combine both.

@umangachapagain if you like, these changes can be merged with /pull/25 and I would abandon this PR.

aruniiird reviewed (commented):
Making it inline with Umanga's PR#25

Commit:
Incorporated new design changes

monstorak-issue-id: monstorak/monstorak/issues/6
monstorak-issue-id: monstorak/monstorak/issues/22
Signed-off-by: Shubhendu <shtripat@redhat.com>

shtripat commented:
@umangachapagain I added the steps for deployment of operator. Still one thing I am not clear is that, where are we taking care of enabling monitoring for new namespace. Can you please have a look comment for required changes in the PR?

@cloudbehl ^^^

umangachapagain commented on docs/Users_guide/index.md:
```diff
@@ -0,0 +1,4 @@
+# User documentation
+
+For now, there's not much to use. Check back later when we have something you
```
We do have something to use.

shtripat replied:
:) Ack. Will remove this statement. Effect of copy + paste ;)

umangachapagain commented on docs/Developers/Design/deploy_procedure_design.md:
```diff
@@ -0,0 +1,65 @@
+An action may be a top-level action and still defined as a prerequisite and the
+caching implementation will ensure that it is executed a maximum of once per
+Procedure execution.
+
```
Line 3 to 13 makes no sense here.

shtripat replied:
Ack. Will remove.

umangachapagain commented on docs/Developers/Design/deploy_procedure_design.md:
```diff
@@ -0,0 +1,65 @@
+
+![ActionGraph](workflow.dot.png)
+
+# StorageMonitoringPlan actions
```
This title is not so meaningful here. Should be  `Prerequisites` and each sub-topic (i.e. left half of the diagram) should be explained.

shtripat replied:
Ack

umangachapagain commented on docs/Developers/Design/deploy_procedure_design.md:
```diff
@@ -0,0 +1,65 @@
+
+## monitoringEnabledForStorage
+
+## deployStorageMixin
```
This is not a sub heading.

shtripat replied:
Ack

umangachapagain commented on docs/Developers/Design/deploy_procedure_design.md:
```diff
@@ -0,0 +1,65 @@
+
+## deployStorageMixin
+
+The monstorak operator consists of two controllers. The first one takes care of
```
Needs a heading. Maybe `Deploy Storage Mixin`. And content should be more about mixin.

shtripat replied:
Ok.

umangachapagain commented on docs/Developers/Design/deploy_procedure_design.md:
```diff
@@ -0,0 +1,65 @@
+## deployStorageMixin
+
+The monstorak operator consists of two controllers. The first one takes care of
+getting the instanciating/reconciating customer resources like StorageCatalog,
```
`instanciating/reconciating` is not a word. Reform the sentence to use the `instantiation` and `reconciliation`

umangachapagain commented on docs/Developers/Design/deploy_procedure_design.md:
```diff
@@ -0,0 +1,65 @@
+second controller forms storage monitoring plans and executes them one by one
+based on how many storage system and what artificats to be deployed.
+
+# Deploy Operator with mixins
```
`Deploy Operator with mixins` is wrong. Should be `Deploy mixins with Operator` and should explain how.

umangachapagain commented on docs/Developers/Design/deploy_procedure_design.md:
```diff
@@ -0,0 +1,65 @@
+To deploy the operator simply deploy to your cluster. The set of steps to be followed
+for deployment of monstorak operator and related artifacts, follow the below steps
+
+## Create the namespace `storage-monitoring`
```
These steps are not specific to design. Should be in the main README file in the project root.

umangachapagain commented on docs/Developers/Design/deploy_procedure_design.md:
````diff
@@ -0,0 +1,65 @@
+
+## Deploy the storage alert rules
+```
+oc create -f jsonnet/manifests/ceph-prometheus-rules.yaml
````
Why do we need this? If we do this, we don't need an operator at all. This step is what the operator will perform internally based on the CRs created.

umangachapagain commented on docs/Developers/Design/CRD_design.md:
```diff
@@ -0,0 +1,116 @@
+the modes of the deployment of storage cluster and the same set of CRDs should
+be used for both the cases.
+
+A given deployment configuration for storage systems contains a Custome Resource
```
`Custome` -> `Custom`

shtripat replied:
Ack

umangachapagain commented on docs/Developers/Design/CRD_design.md:
```diff
@@ -0,0 +1,116 @@
+
+![Hierarchy of monstorak customer resources](crd_hierarchy.dot.png)
+
+# Custom Resources Definitions
```
`Resources` -> `Resource`

umangachapagain commented on docs/Developers/Design/CRD_design.md:
```diff
@@ -0,0 +1,116 @@
+
+This section describes the fields in each of the custom resources.
+
+## **StorageCatelog**
```
Spellcheck

shtripat replied:
Ack

umangachapagain commented on docs/Developers/Design/CRD_design.md:
```diff
@@ -0,0 +1,116 @@
+
+### *Singular: StorageCatalog, Plural:StorageCatalogs*
+
+The StorageCatalog custom resoyrce defines a storage system for which mixins could
```
spellcheck

umangachapagain commented on docs/Developers/Design/CRD_design.md:
```diff
@@ -0,0 +1,116 @@
+      alert: true
+      prometheus:
+      - namespace: "monitoring"
+      - lable: "prometheus:k8s"
```
`lable` -> `label`

shtripat replied:
Ack

umangachapagain commented on docs/Developers/Design/CRD_design.md:
```diff
@@ -0,0 +1,116 @@
+
+### *Singular: StorageDashboard, Plural: StorageDashboards*
+
+The StorageDashboard customer resource defines the dashboards configuration for
```
`customer` -> `custom`

shtripat replied:
Ack :)

umangachapagain commented on docs/Developers/Design/CRD_design.md:
```diff
@@ -0,0 +1,116 @@
+
+The StorageDashboard customer resource defines the dashboards configuration for
+a storage system. This basically captures details like for which version
+dashboards are enabled, if dashboards enabled then which namepsace the
```
spellcheck

shtripat replied:
Got it. Will do that.

umangachapagain commented on docs/Developers/Design/CRD_design.md:
```diff
@@ -0,0 +1,116 @@
+
+### *Singular: StorageMonitoringPlan, Plural: StorageMonitoringPlans*
+
+The StorageMonitoringPlan custom resource defines a plan which shoul be
```
spellcheck

umangachapagain commented on docs/Developers/Design/CRD_design.md:
````diff
@@ -0,0 +1,116 @@
+    - dashboard: true
+```
+
+All CRs live within the `mixins.monstorak.org` group and have version
````
First sentence is fine but should be at the beginning.
Rest is not required.

shtripat replied:
Ack

umangachapagain reviewed (changes requested):
Overall,
1. Deployment steps should be part of README in the root directory.
2. Purpose of `/docs/index.md` is not clear.
3. If we do not have any meaningful content for `/docs/Users_guide/index.md`, it is better we remove this for now.
4. `/docs/Developers/Design/deploy_procedure_design.md` needs to be concise.
5. `/docs/Developers/Design/CRD_design.md` needs some restructuring.

shtripat commented:
> Overall,
> 
>     1. Deployment steps should be part of README in the root directory.

Ack. Will move there.

> 
>     2. Purpose of `/docs/index.md` is not clear.

Will remove this.

> 
>     3. If we do not have any meaningful content for `/docs/Users_guide/index.md`, it is better we remove this for now.

Ack.

> 
>     4. `/docs/Developers/Design/deploy_procedure_design.md` needs to be concise.

Concise in the sense? Deployment steps we are moving to README.md. Any other changes you need?

> 
>     5. `/docs/Developers/Design/CRD_design.md` needs some restructuring.

Commit:
Incorporated review comments

monstorak-issue-id: monstorak/monstorak/issues/6
monstorak-issue-id: monstorak/monstorak/issues/22
Signed-off-by: Shubhendu <shtripat@redhat.com>

cloudbehl commented on README.md:
````diff
@@ -1,3 +1,37 @@
+
+## Create the namespace `storage-monitoring`
+```
+oc create -f deploy/kubernetes/01_namespace.yaml
````
I don't see the kubernetes folder inside deploy right now.

shtripat replied:
Dependent on https://github.com/monstorak/monstorak/pull/24

cloudbehl commented on README.md:
````diff
@@ -1,3 +1,37 @@
+
+## Deploy the custom resources
+```
+oc create -f deploy/kubernets/05_storageAlerts.yaml
````
Typo : /s/kubernets/kubernetes

shtripat replied:
Ack

cloudbehl commented on README.md:
````diff
@@ -1,3 +1,37 @@
+
+## Deploy the storage alert rules
+```
+oc create -f jsonnet/manifests/ceph-prometheus-rules.yaml
````
We are not specific to ceph to can we specify that.

shtripat replied:
Ack

cloudbehl commented on README.md:
```diff
@@ -1,3 +1,37 @@
 # Monitoring operator for storages
 
 This operator takes care of deployment of mixins of storage system in Openshift and Kubernetes. The mixins are the mechanism for adding alerting rules, grafana dashboards and recording rules for prometheus in Openshift and Kubernetes.
```
We can't add dashboard to openshift. So can we just specify the kubernetes.

shtripat replied:
Ack

cloudbehl commented on README.md:
```diff
@@ -1,3 +1,37 @@
+
+# Deploy Operator with mixins
+To deploy the operator simply deploy to your cluster. The set of steps to be followed
+for deployment of monstorak operator and related artifacts, follow the below steps
```
can we simplify the steps? Writing 5 different commands is a tedious task. If we can have single description and add these steps in format that a person can copy paste and done

shtripat replied:
Ack

Commit:
Incoroprated review comments

monstorak-issue-id: monstorak/monstorak/issues/6
monstorak-issue-id: monstorak/monstorak/issues/22
Signed-off-by: Shubhendu <shtripat@redhat.com>

umangachapagain commented:
Being tracked at monstorak/monstorak/pull/31
